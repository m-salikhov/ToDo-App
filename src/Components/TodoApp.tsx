import './todo-list.css';
import { useState } from 'react';
import TodoForm from './TodoForm';
import TodoControl from './TodoControl';
import TodoList from './TodoList';

export interface Todo {
  id: number;
  text: string;
  completed: boolean;
  show: boolean;
}

export default function TodoApp() {
  const [todos, setTodos] = useState<Todo[]>([]);

  return (
    <>
      <h1>todos</h1>

      <div className='todo-wrapper'>
        <TodoForm setTodos={setTodos} />

        <TodoList todos={todos} setTodos={setTodos} />

        <TodoControl setTodos={setTodos} itemsLeft={todos.reduce((acc, todo) => acc + (todo.completed ? 0 : 1), 0)} />
      </div>
    </>
  );
}
