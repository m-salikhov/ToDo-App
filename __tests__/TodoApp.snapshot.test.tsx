import TodoApp from '../src/Components/TodoApp';
import { render } from '@testing-library/react';

test('render todo app', () => {
  const { container } = render(<TodoApp />);
  expect(container).toMatchSnapshot();
});
