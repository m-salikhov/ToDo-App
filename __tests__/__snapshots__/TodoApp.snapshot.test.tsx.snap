// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render todo app 1`] = `
<div>
  <h1>
    todos
  </h1>
  <div
    class="todo-wrapper"
  >
    <form>
      <input
        placeholder="What needs to be done?"
        type="text"
        value=""
      />
      <button
        type="button"
      >
        Add
      </button>
    </form>
    <h4>
      Todo list is empty
    </h4>
    <div
      class="todo-control"
      data-testid="todo-control"
    >
      <p>
        0 items left
      </p>
      <div
        class="todo-control-filter"
      >
        <p
          class="selected"
          id="all"
        >
          All
        </p>
        <p
          id="active"
        >
          Active
        </p>
        <p
          id="completed"
        >
          Completed
        </p>
      </div>
      <p>
        Clear completed
      </p>
    </div>
  </div>
</div>
`;
